src/env.ts

```diff
@@ -1,10 +1,68 @@
-type EnvironmentVariables = {
+import { isDesktop } from '@src/lib/isDesktop'
+import type { EnvironmentConfigurationRuntime } from '@src/lib/constants'
+
+export function generateDomainsFromBaseDomain(baseDomain: string) {
+  return {
+    API_URL: `https://api.${baseDomain}`,
+    SITE_URL: `https://${baseDomain}`,
+    WEBSOCKET_URL: `wss://api.${baseDomain}/ws/modeling/commands`,
+    APP_URL: `https://app.${baseDomain}`,
+  }
+}
+
+export type EnvironmentVariables = {
   readonly NODE_ENV: string | undefined
+  readonly VITE_KITTYCAD_BASE_DOMAIN: string | undefined
   readonly VITE_KITTYCAD_API_BASE_URL: string | undefined
   readonly VITE_KITTYCAD_API_WEBSOCKET_URL: string | undefined
   readonly VITE_KITTYCAD_API_TOKEN: string | undefined
   readonly VITE_KITTYCAD_SITE_BASE_URL: string | undefined
   readonly VITE_KITTYCAD_SITE_APP_URL: string | undefined
+  readonly POOL: string | undefined
```

Comment:
It doesn't have to happen in this PR, but I would like to see a more descriptive name so it's obvious what this variable is controlling, `ENGINE_POOL` perhaps.

Reply:
Sounds, good. I can leave a TODO reminder here.